utilities/transactions/pessimistic_transaction_db.cc

```diff
@@ -588,6 +588,52 @@ Status WritePreparedTxnDB::Get(const ReadOptions& options,
                            &callback);
 }
 
+Iterator* WritePreparedTxnDB::NewIterator(const ReadOptions& options,
```

Comment:
TransactionBaseImpl::GetIterator should also be overridden because it calls NewIterator of db instead of txn db.

Reply:
Good catch!